Source/WebCore/dom/SlotAssignment.cpp

```diff
@@ -522,8 +524,9 @@ void ManualSlotAssignment::slotManualAssignmentDidChange(HTMLSlotElement& slot,
         }
     };
 
-    RenderTreeUpdater::tearDownRenderersAfterSlotChange(*shadowRoot.host());
-    shadowRoot.host()->invalidateStyleForSubtree();
+    RefPtr shadowRootHost = shadowRoot.host();
```

Comment:
Since this is immediately unconditionally dereferenced I think we should use `Ref` instead of `RefPtr`. What do you think?